GVFS/GVFS/CommandLine/UnmountVerb.cs

```diff
@@ -55,22 +54,28 @@ public override void Execute()
                 this.AcquireLock(root);
             }
 
-            if (!this.ShowStatusWhileRunning(
+            string enlistmentId;
+            string mountId;
+            GVFSVerb.ForExistingEnlistment.LookupIds(enlistment, out enlistmentId, out mountId);
+            using (JsonTracer tracer = new JsonTracer(GVFSConstants.GVFSEtwProviderName, "GVFSUnmount", enlistmentId, mountId))
+            {
+                if (!this.ShowStatusWhileRunning(
                 () => { return this.Unmount(root, out errorMessage); },
                 "Unmounting"))
-            {
-                this.ReportErrorAndExit(errorMessage);
-            }
+                {
+                    this.ReportErrorAndExit(errorMessage);
+                }
 
-            if (!this.Unattended && 
-                !this.SkipUnregister &&
-                GVFSPlatform.Instance.SupportsGVFSService)
-            {
-                if (!this.ShowStatusWhileRunning(
-                    () => { return this.UnregisterRepo(root, out errorMessage); },
-                    "Unregistering automount"))
+                if (!this.Unattended &&
+                    !this.SkipUnregister &&
+                    GVFSPlatform.Instance.SupportsGVFSService)
                 {
-                    this.Output.WriteLine("    WARNING: " + errorMessage);
+                    if (!this.ShowStatusWhileRunning(
+                        () => { return this.UnregisterRepo(root, out errorMessage); },
+                        "Unregistering automount"))
+                    {
+                        this.Output.WriteLine("    WARNING: " + errorMessage);
+                    }

```

Comment:
```
Unsuccessful Mounts
  * Mounts that have Error entries
```

Yes, all error entries should be counted as unsuccessful mounts. But that's because every error entry should be followed by an unmount.

```
 * We may choose to ignore some errors in the count. For instance if a user tries to do a clone without 
admin privileges GVFS.exe will log an error when attempting to attach the filter. However, failing to 
attach the filter is expected when GVFS.exe is not running with elevation, and it will request that that 
the Service attach the filter.
```

That is not an error, so please fix that. It used to be an error when it was first written, but now I wouldn't even call it a warning. At most, log it as informational.

Same goes for any other errors - they should be used for things that are truly runtime failures that we can't recover from.

Reply:
@jeschu1 @jrbriggs what are our plans for using this data?  

Is it to drive prioritizing bugs\issues, raise alerts when errors occur, or for something else?

This is probably outside the scope of this PR, but at some point we might need to differentiate between errors that are due to our software\service (e.g. `GVFS.Service` failed to properly enable ProjFS), and failures from user error (e.g. `gvfs mount <path to repo on BitLockered drive>` or `gvfs mount <path to non-existent repo>`).

We can certainly improve the UX for scenarios in the second category, but perhaps they should not be lumped in with errors in the first category?

Reply:
@wilbaker exellent point since we have the error messages we can definitely differentiate which are true errors and which are user errors.  Part of the investigation is fine tuning the kusto query to capture that info correctly.

I'm looking into a fix for this issue mentioned here now.